CMakeLists.txt

```diff
@@ -28,13 +28,72 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-cmake_minimum_required(VERSION 2.8.11)
+cmake_minimum_required(VERSION 3.11)
 
 set (PROJECT smartDeviceLinkCore)
 project (${PROJECT})
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/tools/cmake/modules/")
 
-include(${CMAKE_SOURCE_DIR}/tools/cmake/helpers/platform.cmake)
+include("${CMAKE_CURRENT_LIST_DIR}/cmake/helpers/dependencies.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/tools/cmake/helpers/install_rules.cmake")
+include("${CMAKE_SOURCE_DIR}/tools/cmake/helpers/platform.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/tools/cmake/helpers/build_config_generator.cmake")
+
+include(CMakeDependentOption)
+include(GNUInstallDirs)
+
+if(NOT PLATFORM)
+  set(PLATFORM "Linux")
+endif()
+
+# Request C++11
+#set(CMAKE_CXX_STANDARD 11)
+
+# Disable fallback if standard not supported by compiler
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Enable GNU Extensions
+set(CMAKE_CXX_EXTENSIONS ON)
+
+# Enable PIC as static libraries may be linked to shared objects
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
+if(IS_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/.git")
```

Comment:
@BSolonenko I would extract it to separate cmake file and include it here not to pollute main cmake